common/static/common/js/karma.common.conf.js

```diff
@@ -102,27 +124,60 @@ function junitSettings(config) {
     };
 }
 
-var getPreprocessorObject = function (files) {
-    var preprocessFiles = {};
+var normalizePathsForCoverage = function(files, normalize_func) {
+    var filesForCoverage = {};
 
     files.forEach(function (file) {
-        var pattern = _.isObject(file) ? file.pattern : file;
+        if (_.isObject(file) && file.ignoreCoverage) { return; }
 
-        if (pattern.match(/^common\/js/)) {
-            pattern = path.join(appRoot, '/common/static/' + pattern);
-        } else if (pattern.match(/^xmodule_js\/common_static/)) {
-            pattern = path.join(appRoot, '/common/static/' +
-                pattern.replace(/^xmodule_js\/common_static\//, ''));
+        var pattern = _.isObject(file) ? file.pattern : file;
+        if (normalize_func) {
+            pattern = normalize_func(appRoot, pattern);
+        } else {
+            if (pattern.match(/^common\/js/)) {
+                pattern = path.join(appRoot, '/common/static/' + pattern);
+            } else if (pattern.match(/^xmodule_js\/common_static/)) {
+                pattern = path.join(appRoot, '/common/static/' +
+                  pattern.replace(/^xmodule_js\/common_static\//, ''));
+            }
         }
+        filesForCoverage[pattern] = ['coverage'];
+    });
+
+    return filesForCoverage;
+};
```

Comment:
This logic can be better written if we make a normalize function here in this file a default one.

```javascript
var defaultNormalizeFunc = function (***)  {
  if (pattern.match(/^common\/js/))  {
     pattern = path.join(appRoot, '/common/static/' + pattern);
  } else if (pattern.match(/^xmodule_js\/common_static/)) {
     pattern = path.join(appRoot, '/common/static/' +
     pattern.replace(/^xmodule_js\/common_static\//, ''));
   }
}
```

and then in the `normalizePathsForCoverage` body

```javascript

var normalizePathsForCoverage = function(files, normalizeFn) {
  normalizeFn = normalizeFn || defaultNormalizeFunc;

  // Later
  pattern = normalizeFn(***);
}